src/main/java/perosnal/spotifymatcher/util/HttpRequestSender.java

```diff
@@ -0,0 +1,28 @@
+package perosnal.spotifymatcher.util;
+
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+@Component
+public class HttpRequestSender {
+
+    HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
This should be a `@Bean`. Also scope:

```suggestion
    private final HttpClient httpClient = HttpClient.newHttpClient();
```